src/opencvcamera.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+
+class OpenCVCamera
+{
+public:
+	OpenCVCamera(unsigned int id) : _id(id) , _capture(_id){};
+
+	//caller responsible for cleaning up each capture
+	cv::Mat* Capture()
+	{
+		cv::Mat* mat = new cv::Mat();

```

Comment:
One thing you can do here instead of using pointers is pass in the output matrix to the function.

```
void Capture(cv::Mat& output) {
    _capture >> output;
}
```